### DISCUSSION TITLE
TLS general improvements ideas

### DISCUSSION THREAD

**Discussion by waps11** [hassio-addons/addon-uptime-kuma#124] [Ideas]
I'm unable to send notifications to my HA server as I use a private CA certificate and the server is HTTPS only and so are all my local servers. for now I have to disable TLS verification which is kind of sad. There could be a way to add custom CA so I do not have to a) disable TLS, b) only use public CA. This is for the notifications but would also help on the sensor side as I have the same dilemma.

**Comment by frenck** [comment]
That isn't needed, as this is an add-on, you can talk to HA directly internally.

Use URL: `http://supervisor/core`

../Frenck

**Reply by waps11** [reply]
Thanks for the reply and the solution for the notification part. I haven't been able to replicate it as I get an 401 unauthorized error msg as if my token wasn't recognized by HA. Might be an error on my part tho as I forced HTTPS only on my HAOS installation and this tends to break unexpected things. I'll dig around that problem myself and share here what I found for others.

For the Uptime kuma monitored services issue mentioned on the first post, I'm still unable to securely monitor services that are served under a custom certificate authority's certificate. I have found this thread https://github.com/louislam/uptime-kuma/issues/4015 with a solution but was unable to replicate the solution provided as I use HAOS and it doesn't expose the config options necessary. Under the hood, HA would need to map a folder containing the custom CA to the Uptime Kuma add-on container. IMO, this could be a good general idea and a useful mapping for other add-on as well. For now the workaround is to disable certificate validation which is acceptable in a local lan risk scenario but still is less than ideal.
